File: src/escalonamentoProcessos/fifo.js
import { tempoDeChegadaMinimo, vazio } from "./processo.js"

export function fifo(processos){
    let n = processos.length
    let processosProntos = []
    let retorno = tempoDeChegadaMinimo(processosProntos)
    let it = 0
    let index = 0
    let index_aux = 0

    for(let i = 0; i < n; i++){
        processosProntos[i] = processos[i]
        retorno += processos[i].te
    }

    let escalonado = []

    console.log(`prontos[ ] vazio:\t${vazio(processosProntos)}`)
    console.log(`inseridos prontos[1]:\t${processosProntos[1].id} ${processosProntos[1].tc} ${processosProntos[1].te} ${processosProntos[1].d} ${processosProntos[1].p}\n`)

    while (it != processos.length){
        let aux = processosProntos[it]
        console.log(`index: ${index}\n`)
        for(let i = index; i < index + aux.te; i++){
            escalonado[i] = aux;
            index_aux++;
        }
        index = index_aux;
        it++;
    }

    return escalonado
}

File: escalonamento_processos/fifo.c
//
// Created by kid-a on 20/05/2022.
//

Escalonado fifo(Processo processos[], int n){
    Processo prontos[n];
    int retorno = tempo_de_chegada_minimo(prontos, n);
    int it = 0;
    int index = 0;
    int index_aux = 0;

    for(int i = 0; i < n; i++){
        prontos[i] = processos[i];
        retorno += processos[i].te;
    }

    Escalonado escalonado;
    escalonado.size = retorno;

    /*
    for(int i = 0; i < n; i++){
        prontos[i] = NULL;
    }*/

    printf("prontos[ ] vazio:\t%d\n", vazio(prontos, n));
    printf("inseridos prontos[1]:\t%d %d %d %d %d\n", prontos[1].id, prontos[1].tc, prontos[1].te, prontos[1].d, prontos[1].p);

    while (it != 4){
        Processo aux = prontos[it];
        //aux = porntos[3] = {processo: 3 6 3 10 3}
        printf("index: %d\n", index);
        for(int i = index; i < index + aux.te; i++) {
            escalonado.processos[i] = aux;
            index_aux++;
        }
        index = index_aux;
        it++;
    }


    return escalonado;
}